Upstream: no
Reason: this is a test-compile, a sanity check to make sure the downloaded LLVM works and is able to produce working executables. 
The org.graalvm.toolchain.test/hello file exist but is linked to /usr/x86_64-unknown-linux-gnu instead of /usr/x86_64-pc-linux-gnu.
No problem with other binaries. To be confirmed, but it looks like the “x86_64-unknown-linux-gnu” target is hardcoded in many test files.
Ongoing discussions on the GraalVM Slack: https://graalvm.slack.com/archives/CR90E12KT/p1744299360359589

From c4f04676f7f41a95d52913c84b9f64e757c0e6e8 Mon Sep 17 00:00:00 2001
From: s0dyy <maxime.sorin@clever-cloud.com>
Date: Thu, 10 Apr 2025 21:17:31 +0200
Subject: [PATCH] no such file or directory: mxbuild/sdk/linux-amd64/src/org.graalvm.toolchain.test/hello

---
 sdk/mx.sdk/suite.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/sdk/mx.sdk/suite.py b/sdk/mx.sdk/suite.py
index 20806182124..711d12b50ed 100644
--- a/sdk/mx.sdk/suite.py
+++ b/sdk/mx.sdk/suite.py
@@ -442,11 +442,6 @@ suite = {
       "jacoco" : "exclude",
       "testProject" : True,
     },
-    "org.graalvm.toolchain.test" : {
-      "class" : "ToolchainTestProject",
-      "subDir" : "src",
-      "buildDependencies" : ["LLVM_TOOLCHAIN"],
-    },
     "org.graalvm.shadowed.org.jline": {
       # shaded JLINE_*
       "subDir": "src",
@@ -1143,7 +1138,6 @@ ML=<path:LLVM_TOOLCHAIN>\\bin\\llvm-ml
       },
       "dependencies": [
         "LLVM_TOOLCHAIN",
-        "org.graalvm.toolchain.test",
       ],
     },
   },
-- 
2.48.1

